sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTestCommon.java

```diff
@@ -460,6 +446,41 @@ void testWriteWithErrorsReturnedAllowedErrors() throws Exception {
     pipeline.run();
   }
 
+  void testWriteWithElasticClientResponseException() throws Exception {
+    Write write =
+        ElasticsearchIO.write()
+            .withConnectionConfiguration(connectionConfiguration)
+            .withMaxBatchSize(1000L)
+            .withMaxBatchSizeBytes(
+                Long.MAX_VALUE) // Max long number to make sure it flushes 2 docs.
+            .withThrowWriteErrors(false)
+            .withIdFn(new ExtractValueFn("id"));
+
+    List<String> data =
+        ElasticsearchIOTestUtils.createDocuments(
+            2, ElasticsearchIOTestUtils.InjectionMode.INJECT_ONE_ID_TOO_LONG_DOC_AT_THE_END);
+
+    PCollectionTuple outputs = pipeline.apply(Create.of(data)).apply(write);
+    PCollection<String> success =
+        outputs
+            .get(Write.SUCCESSFUL_WRITES)
+            .apply("Convert success to input ID", MapElements.via(mapToInputIdString));
+
+    PCollection<String> fail =
+        outputs
+            .get(Write.FAILED_WRITES)
+            .apply("Convert fails to input ID", MapElements.via(mapToInputIdString));
+
+    Set<String> validIds = new HashSet<>();
+    validIds.add("0");
+    PAssert.that(success).containsInAnyOrder(validIds);
```

Comment:
nit: could be slightly simplified to `PAssert.that(success).containsInAnyOrder(Collections.singleton("0"))`

Reply:
Although this point is likely moot once you achieve having both docs in a single bulk request.